### DISCUSSION TITLE
Grafana mimir filter by label in datasource

### DISCUSSION THREAD

**Discussion by osipovdaniil** [grafana/mimir#5854] [Help and support] [answered]
Hello! I have datasource for grafana-mimir-query-frontend. Into grafana mimir push metrics from multiple cluster
For example metrics `kube_pod_label{cluster="test1", tenant="system"}`, `kube_pod_label{cluster="test2", tenant="system"}`
I want to have datasource for tenant system and filter labels by `cluster=test1`. It's possible?
For example VictoriaMetrics has extra_label for filter in datasource. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1021#issuecomment-762145324)

**Comment by wltbenade** [comment]
You can make use of the custom query parameters block under your datasource config, setting it as `cluster=test1` or as needed.
![image](https://github.com/grafana/mimir/assets/66667751/93f09a96-369b-4d61-8a89-11ca21a789a9)

I should note that this would mean that ALL metrics coming from that cluster needs to have that label, or they won't be picked up.

**Reply by dimitarvdimitrov** [reply, marked as answer]
that's not functionality that the [prometheus query API supports](https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries) and by extention Mimir's query API doesn't support it
